src/test/javascript/spec/service/orion/orion-connector.service.spec.ts

```diff
@@ -0,0 +1,240 @@
+import { TestBed } from '@angular/core/testing';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { OrionConnectorService } from 'app/shared/orion/orion-connector.service';
+import { ArtemisTestModule } from '../../test.module';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { MockProvider } from 'ng-mocks';
+import { Injector } from '@angular/core';
+import { MockRouter } from '../../helpers/mocks/mock-router';
+import { Router } from '@angular/router';
+import { ExerciseView, OrionBuildConnector, OrionExerciseConnector, OrionSharedUtilConnector, OrionState, OrionVCSConnector } from 'app/shared/orion/orion';
+import { Annotation } from 'app/exercises/programming/shared/code-editor/ace/code-editor-ace.component';
+import { AlertService } from 'app/core/util/alert.service';
+import { REPOSITORY } from 'app/exercises/programming/manage/code-editor/code-editor-instructor-base-container.component';
+import { Feedback } from 'app/entities/feedback.model';
+
+describe('OrionConnectorService', () => {
+    let serviceUnderTest: OrionConnectorService;
+    const router = new MockRouter();
+    let alertService: AlertService;
+
+    const exercise = { id: 42 } as ProgrammingExercise;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            providers: [
+                OrionConnectorService,
+                MockProvider(Injector),
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: Router, useValue: router },
+            ],
+        });
+
+        serviceUnderTest = TestBed.inject(OrionConnectorService);
+        alertService = TestBed.inject(AlertService);
+
+        // Mock all connectors. It may or may not be possible to do this automatically
```

Comment:
I'm not sure how I should feel about this comment. Could you elaborate? "May or may not" 🤷

Reply:
Well, the simple truth is I tried to automate it for some time and then gave up. This comment is only there to tell later readers that this may or may not is the best solution; If someone is to copy it, they should probably check if they find a better solution